### PR TITLE
Refactor auth backend

### PR DESCRIPTION
**Pull request by lunify**
- standardizes server responses to better align with http and rest conventions, mainly when it comes to http statuses and error responses, making them more easily consumed by the frontend.
- fixes a couple of bugs

### CODE REVIEW

File: src/utils/auth.js [hunk 1]
```diff
@@ -68,61 +50,65 @@ async function getLoggedUser(req, res) {
...
+      }
     }
 
-    if (error.errors && error.errors.password) {
-      return res.status(400).json({ message: error.errors.password.message })
-    }
-    // in case any errors are not properly handled.
-    return res.status(400).json({ message: error })
+    // other errors would be server generated errors
+    return res.status(500).send()
```
Comment (SebastianLF) [inline src/utils/auth.js:118]: Great practice!

File: src/utils/auth.js [hunk 2]
```diff
@@ -131,3 +117,23 @@ module.exports = {
...
+function getCookieOpts() {
+  const afterThirtyDays = new Date(Date.now() + 30 * 24 * 60 * 60)
+
+  const cookieOptions = {
+    httpOnly: true,
+    expires: afterThirtyDays,
+    signed: true
+  }
+  return cookieOptions
+}
```
Comment (SebastianLF) [inline src/utils/auth.js]: What's the point of setting cookie expiration to 30 days while jwt token expires after 2h?

Comment (lunify) [reply src/utils/auth.js]: Great catch. That's a valid question and needs to be changed to synchronize the cookie with the token.

Besides that, there's a prevalent practice using refresh tokens but I didn't look into that tbh. It's a possible improvement to implement.

Comment (SebastianLF) [reply src/utils/auth.js]: For now, it doesnt matter anyway. A new cookie will generated when jwt is outdated. Right?

Comment (lunify) [reply src/utils/auth.js]: indeed, since the user will have to login manually. But I went ahead and synchronized them anyway

Comment (SebastianLF) [reply src/utils/auth.js]: Great!

**Review by SebastianLF — APPROVED** [review]

**Pull request merged**